packages/server/api/src/app/flows/flow/flow-version.controller.ts

```diff
@@ -110,30 +111,38 @@ export const flowVersionController: FastifyPluginAsyncTypebox = async (
     },
   );
 
-  fastify.get('/:id/test-output', GetFlowTestOutputRequestOptions, (request) =>
-    request.query.stepIds.reduce<Record<string, unknown>>((acc, stepId) => {
-      acc[stepId] = {};
-      return acc;
-    }, {}),
-  );
-};
+  fastify.get(
+    '/:flowVersionId/test-output',
+    {
+      config: {
+        allowedPrincipals: [PrincipalType.USER],
+      },
+      schema: {
+        params: Type.Object({
+          flowVersionId: Type.String(),
+        }),
+        tags: ['flow-step-test-output'],
+        description: 'Gets the test output for a flow version',
+        security: [SERVICE_KEY_SECURITY_OPENAPI],
+        querystring: Type.Object({
+          stepIds: Type.Array(Type.String()),
+        }),
+      },
+    },
+    async (request): Promise<Record<OpenOpsId, unknown>> => {
+      const { stepIds } = request.query;
+      const { flowVersionId } = request.params;
 
-const GetFlowTestOutputRequestOptions = {
-  config: {
-    allowedPrincipals: [PrincipalType.USER],
-    permission: Permission.READ_FLOW,
-  },
-  schema: {
-    tags: ['flow-version'],
-    description:
-      'Get flow test output by flowVersionId. Optionally, filter by stepIds',
-    params: Type.Object({
-      id: OpenOpsId,
-    }),
-    querystring: Type.Object({
-      stepIds: Type.Array(Type.String({})),
-    }),
-  },
+      const outputs = await flowStepTestOutputService.list({
+        stepIds,
+        flowVersionId,
+      });
+
+      return Object.fromEntries(
+        outputs.map(({ stepId, output }) => [stepId, output]),
+      );
```

Comment:
We also need the lastTestDate (I guess the `lastUpdated` timestamp from the DB).

Reply:
done